### PR TITLE
added anchoring

### PR DESCRIPTION
**Pull request by Karollus**
@Avsecz 
To make the review eaiser, there is an ipynb with some testing code + example files. This should probably be removed before merging

Best,
Alex

### CODE REVIEW

File: kipoiseq/dataloaders/sequence.py [hunk 1]
```diff
@@ -381,3 +383,52 @@ def get_output_schema(cls):
...
         return output_schema
+    
+class AnchoredGTFDl(Dataset):
+    
+    def __init__(self, gtf_file, fasta_file, 
+                 num_upstream, num_downstream,
+                 gtf_filter, anchor_extractor,
+                 transform,
+                 interval_attrs,
+                 use_strand=True):
```
Comment (Avsecz) [inline kipoiseq/dataloaders/sequence.py:394]: Please add a docstring describing different parameters

File: kipoiseq/dataloaders/sequence.py [hunk 2]
```diff
@@ -381,3 +383,52 @@ def get_output_schema(cls):
...
+        gtf = pr.read_gtf(gtf_file).df
+        gtf = gtf_filter(gtf)  # Optional
```
Comment (Avsecz) [inline kipoiseq/dataloaders/sequence.py:396]: allow gtf_filter to be a string (representing the argument for pd.DataFrame.query) or a function. If None, skip the filtering step.

File: kipoiseq/dataloaders/sequence.py [hunk 3]
```diff
@@ -381,3 +383,52 @@ def get_output_schema(cls):
             output_schema.targets = None
 
         return output_schema
+    
+class AnchoredGTFDl(Dataset):
+    
+    def __init__(self, gtf_file, fasta_file, 
+                 num_upstream, num_downstream,
+                 gtf_filter, anchor_extractor,
```
Comment (Avsecz) [inline kipoiseq/dataloaders/sequence.py:391]: Rename anchor_extractor to `anchor` and allow it to be a string or a function. If string, please hard-code and implemnet anchor/anchor_extractor for a key points on the mRNA such as: TSS, start_codon, stop_codon, polya

**Review by Avsecz — CHANGES_REQUESTED** [review]
Thanks. A few minor suggestions to polish the API.

File: kipoiseq/dataloaders/sequence.py (hunk 3 above)
Comment (Karollus) [reply kipoiseq/dataloaders/sequence.py]: Done, not 100% sure on the polyA one. It does not seem to be a feature in standard gtf, so I infer it as the gene end.

File: kipoiseq/dataloaders/sequence.py (hunk 2 above)
Comment (Karollus) [reply kipoiseq/dataloaders/sequence.py]: Done

File: kipoiseq/dataloaders/sequence.py (hunk 1 above)
Comment (Karollus) [reply kipoiseq/dataloaders/sequence.py:401]: Done

**Review by Avsecz — APPROVED** [review]
Very elegant. Great job!

**Comment by haimasree** [comment]
@Karollus Hi Alex, can you please merge this branch to master now that it is approved?

**Comment by Karollus** [comment]
@haimasree As far as I can see, i do not have the priviledges to merge

Screenshot from 2021-05-14 20-33-52

**Comment by haimasree** [comment]
Okay I merged it.

**Pull request merged**